src/main/client/system/serverKeypress.ts

```diff
@@ -61,10 +69,39 @@ function handleKeyDown(key: number) {
         return;
     }
 
+    if (!keyStates[key]) {
+        keyStates[key] = { isPressed: false, pressStartTime: 0, pressDuration: 0 };
+    }
+
+    keyStates[key].isPressed = true;
+    keyStates[key].pressStartTime = Date.now();
     nextValidDownPress[key] = Date.now() + 500;
     alt.emitServer(Events.systems.keypress.invokeDown, key);
 }
 
 alt.onServer(Events.systems.keypress.update, handleUpdate);
 alt.on('keyup', handleKeyup);
 alt.on('keydown', handleKeyDown);
+
+
+alt.everyTick(() => {
+    const currentTime = Date.now();
+
+    for (const key in keyStates) {
+        const state = keyStates[key];
+
+        if (state.isPressed) {
+            state.pressDuration = currentTime - state.pressStartTime;
+
+            if (state.pressDuration >= requiredPressDuration) {
+                state.isPressed = false;
+                state.pressStartTime = 0;
+                state.pressDuration = 0;
+                alt.emitServer(Events.systems.keypress.invokeHold, parseInt(key));
+            }
+        } else {
+            state.pressStartTime = 0;
+            state.pressDuration = 0;
+        }
```

Comment:
Just a small change here, I think the rest looks good. Just don't want to nest if statements into oblivion.

```suggestion
        if (!state.isPressed) {
            state.pressStartTime = 0;
            state.pressDuration = 0;
            continue;
        }
        
        state.pressDuration = currentTime - state.pressStartTime;
        if (state.pressDuration < requiredPressDuration) {
            continue;
        }
        
        state.isPressed = false;
        state.pressStartTime = 0;
        state.pressDuration = 0;
        alt.emitServer(Events.systems.keypress.invokeHold, parseInt(key));
```